animal-sort/animalSort.js

```diff
@@ -0,0 +1,21 @@
+function animalSort (collection, iteratees = []) {
+  if (collection === null) {
+    return []
+  }
+  if (!iteratees.length) {
+    iteratees = ['numberOfLegs', 'name']
+  }
+  let compareBy = getMultipleIteratees(iteratees)
+
+  return collection.sort((val1, val2) => eval(compareBy))
```

Comment:
The comparator method could have been much simpler. Something like this would have also worked:
```
{
  if (a.numberOfLegs > b.numberOfLegs) {
    return 1;
  } if (a.numberOfLegs === b.numberOfLegs) {
    if (a.name > b.name) {
      return 1;
    }
  }
  return -1;
}
```